### PR TITLE
Add s390x and disable osx-arm64

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -25,19 +54,33 @@ chmod og-w COPYING*
...
 cd build
 
-if [[ "$target_platform" == "osx-64" ]]; then
-  export CPPFLAGS="$CPPFLAGS -mmacosx-version-min=${MACOSX_DEPLOYMENT_TARGET}"
-  export CFLAGS="$CFLAGS -mmacosx-version-min=${MACOSX_DEPLOYMENT_TARGET}"
-  export CXXFLAGS="$CXXFLAGS -mmacosx-version-min=${MACOSX_DEPLOYMENT_TARGET}"
-  export LDFLAGS="$LDFLAGS -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs"
+if [[ "$target_platform" == osx-arm64 ]]; then
+  OSX_ARCH="arm64"
+elif [[ "$target_platform" -- osx-64 ]]; then
```
Comment (beckermr) [inline recipe/build.sh:59]:
```suggestion
elif [[ "$target_platform" == osx-64 ]]; then
```

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

File: recipe/build.sh (hunk 1 above)
Comment (isuruf) [reply recipe/build.sh]: Thanks. Pushed a commit before I saw your comment.

Comment (beckermr) [reply recipe/build.sh]: :)

**Comment by beckermr** [comment]
What was broken in osx-arm64? Was it just that we were not making the binutils at all?

**Comment by isuruf** [comment]
It wasn't cross compiling. Just building for macos-x86_64.

**Pull request merged**